docs/src/writing-constructs.rst

```diff
@@ -0,0 +1,208 @@
+.. Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+   This work is licensed under a Creative Commons Attribution-NonCommercial-ShareAlike 4.0
+   International License (the "License"). You may not use this file except in compliance with the
+   License. A copy of the License is located at http://creativecommons.org/licenses/by-nc-sa/4.0/.
+
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+   either express or implied. See the License for the specific language governing permissions and
+   limitations under the License.
+
+.. _writing_constructs:
+
+#########################
+Writing |cdk| Constructs
+#########################
+
+This topic provides some tips writing idiomatic new constructs for the |cdk|.
+The tips here apply equally to constructs written for inclusion in the AWS
+Construct Library, purpose-built constructs to achieve a well-defined goal,
+or constructs that serve as building blocks for assembling your cloud
+applications.
+
+General Design Priciples
```

Comment:
I am not sure about the distinction between the general principles and the implementation details. Some items in this section doesn't fit the "general design principles" category in my mind and vice versa. I wonder if we should just provide a single list...